src/Model/GetSourceSelectionResultFromOrder.php

```diff
@@ -89,7 +99,7 @@ private function getSelectionRequestItems(iterable $orderItems): array
     {
         $selectionRequestItems = [];
         foreach ($orderItems as $orderItem) {
-            if ($orderItem->isDummy()) {
+            if (!$this->isSourceItemManagementAllowedForProductType->execute($orderItem->getProductType())) {
```

Comment:
@kernstmediarox Does this function cover off `isDummy()`? Or would it be safer to do this like ?

```diff
  foreach ($orderItems as $orderItem) {
-   if ($orderItem->isDummy()) {
+   if ($orderItem->isDummy() || !$this->isSourceItemManagementAllowedForProductType->execute($orderItem->getProductType())) {
```

Reply:
In our opinion the method "isDummy" is not necessary to validate products for deduction. If you check this example: Magento\InventorySales\Model\PlaceReservationsForSalesEvent Line 112
you can see, it's enough to use only this method to validate the products

Reply:
Thanks for the feedback, this will help in our review.

We appreciate the PR and the effort you've gone to here.